### ISSUE TITLE
Proxy environment variable no longer works

### ISSUE THREAD

**Issue by erkexzcx** [erkexzcx/stoppropaganda#66]
Since I switched from net/http to fasthttp, proxy functionality has changed/gone. Anyone have time to dig and see if it's possible to bring it back? :)

Environment variable: https://github.com/erkexzcx/stoppropaganda/blob/main/docker-compose-tor.yaml#L15

Used to be in code here: https://github.com/erkexzcx/stoppropaganda/blob/v0.0.25/internal/stoppropaganda/stoppropaganda.go#L56

And it's gone in current implementation.

**Comment by deputinizer** [comment]
Should work similarly:

https://github.com/erkexzcx/stoppropaganda/pull/76

Also from environment:
https://github.com/deputinizer/stoppropaganda/commit/ec6731486754ddaa415f3c444caf3fd6b0aabae6?diff=unified#diff-2bdeed6169330787e0b2ed4245ce6908c32357c12b62dce20a5b3c2c25494ed2R67

**Comment by erkexzcx** [comment]
@deputinizer would be great if you could send a PR with updated README instructions as well as necesarry modification in code. :)

**Comment by erkexzcx** [comment]
Also we need to note that Proxy does not work with DNS DOS. It's only applicalbe to websites DOS.

**Comment by erkexzcx** [comment]
> FastHttp already caches DNS btw.

I want to believe this, but currently I am getting way, waaaaaaay better CPU performance than without existing caching. Could it be that it fasthttp did not cache them?

**Comment by deputinizer** [comment]
Just look into it with pprof:

```golang
import (
	"net/http"
	_ "net/http/pprof"
)
```
Then somewhere in `Start()`:

```golang
log.Println(http.ListenAndServe("127.0.0.1:6060", nil))
```

```bash
go tool pprof http://localhost:6060/debug/pprof/profile?seconds=30
# after 30 seconds:
top
svg
```

**Comment by deputinizer** [comment]
> Environment variable: https://github.com/erkexzcx/stoppropaganda/blob/main/docker-compose-tor.yaml#L15

Then it goes like this:
```yaml
    environment:
      SP_WORKERS: "1000"
      SP_TIMEOUT: "10s"
      SP_DNSWORKERS: "100"
      SP_DNSTIMEOUT: "1s"
      SP_USERAGENT: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36"
      all_proxy: "socks5://tor:9050"
```

**Comment by erkexzcx** [comment]
Should be fixed in v0.0.34
